src/hotspot/share/compiler/oopMap.inline.hpp

```diff
@@ -68,7 +68,7 @@ void OopMapDo<OopFnT, DerivedOopFnT, ValueFilterT>::iterate_oops_do(const frame
   #ifndef COMPILER2
       COMPILER1_PRESENT(ShouldNotReachHere();)
```

Comment:
As I noted in a private comment, I think this logic is simply wrong for JVMCI but since we never build JVMCI without C2 we never encounter it.  Derived oops should only be encountered if C2 in available or if EnableJVMCI is true.  I don't really understand the `COMPILER1_PRESENT` guard here either.  It seems like it should be more like:
```
#ifndef COMPILER2
#if INCLUDE_JVMCI
      if (!EnableJVMCI)
#endif
        ShouldNotReachHere();
#endif // !COMPILER2
```

Reply:
Done.